activestorage/active.py

```diff
@@ -111,16 +191,48 @@ def __from_storage(self, stripped_indexer, drop_axes, out_shape, out_dtype, fsre
             self._process_chunk(fsref, chunk_coords,chunk_selection, out, out_selection,
                                     drop_axes=drop_axes)
 
-        if self.method is not None:
-            return self.method(out)
-        else:
-            return out
+        if method is not None:
+            out = method(out)
+            if self._components:
+                # Return a dictionary of components containing the
+                # reduced data and the sample size ('n'). (Rationale:
+                # cf-python needs the sample size for all reductions;
+                # see the 'mtol' parameter of cf.Field.collapse.)
+                #
+                # Note that in this case the reduced data must always
+                # have the same number of dimensions as the original
+                # array, i.e. 'drop_axes' is always considered False,
+                # regardless of its setting. (Rationale: dask
+                # reductions require the per-dask-chunk partial
+                # reductions to retain these dimensions so that
+                # partial results can be concatenated correctly.)
+                n = prod(out_shape)
+                shape1 = (1,) * len(out_shape)
+                n = np.reshape(n, shape1)
+                out = out.reshape(shape1)
+
+                if self._method == "mean":
+                    out = {"total": out, "n": n}
+                else:
+                    out = {self._method: out, "n": n}
+            elif self._method == "mean":
```

Comment:
hang on a sec, this if/elif is a bit strange - also you are checking for `self._method == "mean"` inside the if case above

Reply:
It is a bit confusing :). It's the same as:
```python
        if self._components:
            # Return the result components dictionary
            <stuff>
        else:
            # Return a numpy array of the result
            if self._method == "mean":
                # In this case, 'out' is sum, so needs divding by the
                # sample size.
                <other stuff>
            else:
                pass
```
I'll add some more comments ...

Reply:
thanky, sir!